srcCxx/ShapeType_main.cxx

```diff
@@ -292,30 +354,26 @@ int run(DomainId_t domain_id, bool use_security,
     int sent_count = 0;
     int recv_count = 0;
     while  ( exit_application == false ) {
-        printf("\nLoop: wait count = %d, sent count = %d, "
-	       "received count = %d\n",
-	       wait_count++, sent_count, recv_count);
+        char msg[256];
+        snprintf(msg, 256, "\nLoop: wait count = %d, sent count = %d, "
+	         "received count = %d\n",
+	         wait_count, sent_count, recv_count);
+        logger.log_message(msg, Verbosity::DEBUG);
+        
+        wait_count++;
         wait_set->wait(active_cond, wait_timeout);
 
         if ( reader != NULL ) {
-#if defined(TWINOAKS_COREDX)
-            while ( reader->take_next_sample(&shape, &info) == DDS_RETCODE_OK )
-#else
             while ( reader->take_next_sample(shape, info) == DDS_RETCODE_OK )
-#endif
               {
-                printf("Received sample...\n");
                 if ( info.valid_data ) {
                     recv_count++;
-#if defined(TWINOAKS_COREDX)
-                    ShapeType::print(stdout, &shape);
-#elif defined(INTERCOM_DDS)
-                    std::cout << shape << std::endl;
-#elif defined OPENDDS
-                    ShapeTypeConfigurator::print_data(shape);
-#else
-                    ShapeTypeTypeSupport::print_data(&shape);
```

Comment:
This change makes [print_data](https://github.com/omg-dds/dds-security/blob/35a36a8329036dcf464504164b643d9d6d00f100/srcCxx/opendds/ShapeType_configurator_opendds.h#L65) unused. Should we delete it and include @mitza-oci in the review?

Reply:
Good point! 
I'll add Adam.